app/src/main/java/com/rightpoint/emojipuzzler/puzzle/PuzzleViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.rightpoint.emojipuzzler.puzzle
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.rightpoint.domain.IDomainClient
+import com.rightpoint.domain.models.EmojiPuzzle
+import com.rightpoint.emojipuzzler.EmojiPuzzleApplication
+import kotlinx.coroutines.*
+
+class PuzzleViewModel: ViewModel() {
+
+    private var domain: IDomainClient = EmojiPuzzleApplication.domainClient
+
+    var puzzle: EmojiPuzzle? = null
```

Comment:
#nth - "puzzle" should be private since you are only using it inside of PuzzleViewModel